src/SixLabors.Fonts/FontCollectionExtensions.cs

```diff
@@ -17,12 +19,31 @@ public static FontCollection AddSystemFonts(this FontCollection collection)
         {
             // This cast is safe because our underlying SystemFontCollection implements
             // both interfaces separately.
-            foreach (FontMetrics? metric in (IReadOnlyFontMetricsCollection)SystemFonts.Collection)
+            foreach (FontMetrics metric in (IReadOnlyFontMetricsCollection)SystemFonts.Collection)
             {
                 ((IFontMetricsCollection)collection).AddMetrics(metric);
             }
 
             return collection;
         }
+
+        /// <summary>
+        /// Adds the fonts from the <see cref="SystemFonts"/> collection to this <see cref="FontCollection"/>.
+        /// </summary>
+        /// <param name="collection">The font collection.</param>
+        /// <param name="match">The <see cref="Predicate{T}"/> delegate that defines the conditions of <see cref="FontMetrics"/> to add into the font collection.</param>
+        /// <returns>The <see cref="FontCollection"/> containing the system fonts.</returns>
+        public static FontCollection AddSystemFonts(this FontCollection collection, Predicate<FontMetrics> match)
```

Comment:
We'll need to add a unit test for this overload. Maybe one that checks that filters by culture and checks the collection.

Reply:
There has already been an unit test for `AddSystemFonts()`. So just filter all the system fonts and check if the font collection is empty,

`collection.AddSystemFonts(p => false);`

I think it's enough to make sure that the custom filter works in the method.